batch/batch/front_end/front_end.py

```diff
@@ -444,14 +459,19 @@ async def _get_full_job_status(app, record):
 
     batch_id = record['batch_id']
     job_id = record['job_id']
-    attempt_id = record['attempt_id']
     state = record['state']
     format_version = BatchFormatVersion(record['format_version'])
 
-    if state in ('Pending', 'Creating', 'Ready', 'Cancelled'):
+    if state in ('Pending', 'Creating', 'Ready'):
+        return None
+
+    if state == 'Cancelled' and record['last_cancelled_attempt_id'] is None:
         return None
 
-    if state in ('Error', 'Failed', 'Success'):
+    attempt_id = record['attempt_id'] or record['last_cancelled_attempt_id']
+    assert attempt_id
```

Comment:
prefer is not None because if we ever make attempt ids that are false-y this will fail.